include/all-io.h

```diff
@@ -18,6 +18,7 @@
 #endif
 
 #include "c.h"
+#include "xalloc.h"
```

Comment:
And this is also bad in another way, some tools define XALLOC_EXIT_CODE before `include xalloc.h` if you include the file in another header, you will see collisions:
https://download.copr.fedorainfracloud.org/results/packit/util-linux-util-linux-2081/fedora-rawhide-x86_64/05582964-util-linux/builder-live.log.gz

xalloc.h should not be used in lib/  (hmm ... for example lib/logindefs.c uses it, seems like a bad idea, for example for agetty. Seems like an opportunity for another PR :-)